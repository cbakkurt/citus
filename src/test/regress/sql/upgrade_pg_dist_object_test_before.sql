-- create some objects that we just included into distributed object
-- infrastructure in 9.1 versions but not included in 9.0.2

-- extension propagation --

-- create an extension on all nodes and a table that depends on it
CREATE EXTENSION isn;
SELECT run_command_on_workers($$CREATE EXTENSION isn;$$);

CREATE TABLE isn_dist_table (key int, value issn);
SELECT create_reference_table('isn_dist_table');

-- create an extension on all nodes, but do not create a table depending on it
CREATE EXTENSION seg;
SELECT run_command_on_workers($$CREATE EXTENSION seg;$$);

-- schema propagation --

-- public schema
CREATE TABLE dist_table (a int);
SELECT create_reference_table('dist_table');

-- custom schema
CREATE SCHEMA new_schema;

SET search_path to new_schema;

CREATE TABLE another_dist_table (a int);
SELECT create_reference_table('another_dist_table');
