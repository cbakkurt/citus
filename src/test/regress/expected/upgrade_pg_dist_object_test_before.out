-- create some objects that we just included into distributed object
-- infrastructure in 9.1 versions but not included in 9.0.2
-- extension propagation --
-- create an extension on all nodes and a table that depends on it
CREATE EXTENSION isn;
SELECT run_command_on_workers($$CREATE EXTENSION isn;$$);
         run_command_on_workers
---------------------------------------------------------------------
 (localhost,57636,t,"CREATE EXTENSION")
 (localhost,57637,t,"CREATE EXTENSION")
(2 rows)

CREATE TABLE isn_dist_table (key int, value issn);
SELECT create_reference_table('isn_dist_table');
 create_reference_table
---------------------------------------------------------------------

(1 row)

-- create an extension on all nodes, but do not create a table depending on it
CREATE EXTENSION seg;
SELECT run_command_on_workers($$CREATE EXTENSION seg;$$);
         run_command_on_workers
---------------------------------------------------------------------
 (localhost,57636,t,"CREATE EXTENSION")
 (localhost,57637,t,"CREATE EXTENSION")
(2 rows)

-- schema propagation --
-- public schema
CREATE TABLE dist_table (a int);
SELECT create_reference_table('dist_table');
 create_reference_table
---------------------------------------------------------------------

(1 row)

-- custom schema
CREATE SCHEMA new_schema;
SET search_path to new_schema;
CREATE TABLE another_dist_table (a int);
SELECT create_reference_table('another_dist_table');
 create_reference_table
---------------------------------------------------------------------

(1 row)

